common/src/main/java/org/opensearch/ml/common/connector/McpConnector.java

```diff
@@ -205,21 +207,37 @@ protected Map<String, String> createDecryptedHeaders(Map<String, String> headers
     }
 
     @Override
-    public void decrypt(String action, BiFunction<String, String, String> function, String tenantId) {
-        Map<String, String> decrypted = new HashMap<>();
+    public void decrypt(String action, BiFunction<String, String, Future<String>> function, String tenantId) {
+        Map<String, Future<String>> decryptingTempCredential = new HashMap<>();
+        decryptedCredential = new HashMap<>();
         for (String key : credential.keySet()) {
-            decrypted.put(key, function.apply(credential.get(key), tenantId));
+            decryptingTempCredential.put(key, function.apply(credential.get(key), tenantId));
         }
-        this.decryptedCredential = decrypted;
+        fillCredential(decryptingTempCredential, decryptedCredential);
         this.decryptedHeaders = createDecryptedHeaders(headers);
     }
 
+    private void fillCredential(Map<String, Future<String>> decrypted, Map<String, String> decryptedCredential) {
```

Comment:
Input validation?
    if (decrypted == null || decryptedCredential == null) {
        throw new IllegalArgumentException("Input maps cannot be null");
    }

Reply:
This is not part of latest changes.